src/secrets/CommandMkdir.ts

```diff
@@ -50,16 +56,57 @@ class CommandMkdir extends CommandPolykey {
           },
           logger: this.logger.getChild(PolykeyClient.name),
         });
-        await binUtils.retryAuthentication(
-          (auth) =>
-            pkClient.rpcClient.methods.vaultsSecretsMkdir({
-              metadata: auth,
-              nameOrId: secretPath[0],
-              dirName: secretPath[1],
-              recursive: options.recursive,
-            }),
-          meta,
+        const response = await binUtils.retryAuthentication(async (auth) => {
+          const response =
+            await pkClient.rpcClient.methods.vaultsSecretsMkdir();
+          const writer = response.writable.getWriter();
+          let first = true;
+          for (const [vault, path] of secretPaths) {
+            await writer.write({
+              nameOrId: vault,
+              dirName: path,
+              metadata: first
+                ? { ...auth, options: { recursive: options.recursive } }
+                : undefined,
+            });
+          }
+          await writer.close();
+          return response;
+        }, meta);
```

Comment:
You're never updating `first` to `false`. It won't break things given how metadata works, however it will be sending extra information in each message.

Reply:
To avoid setting the variable to false each iteration, I have done the following which is more verbose, but avoids this issue. Should I change this and keep it simple?

```ts
if (first) {
  first = false;
  await writer.write({
    nameOrId: vault,
    dirName: path,
    metadata: {
      ...auth,
      options: { recursive: options.recursive },
    },
  });
} else {
  await writer.write({ nameOrId: vault, dirName: path });
}
```

Reply:
Updating a boolean every iteration of a loop is not a problem. Don't worry about micro-optimisations like that. Just keep the code here clean.